ext/src/Cassandra/Date.c

```diff
@@ -97,14 +100,51 @@ PHP_METHOD(Date, toDateTime)
   datetime_obj = zend_object_store_get_object(datetime TSRMLS_CC);
 #endif
 
-  str_len = spprintf(&str, 0, "%lld", cass_date_time_to_epoch(self->date, 0));
+  str_len = spprintf(&str, 0, "%lld",
+                     cass_date_time_to_epoch(self->date,
+                                             time_obj != NULL ? time_obj->time : 0));
   php_date_initialize(datetime_obj, str, str_len, NULL, NULL, 0 TSRMLS_CC);
   efree(str);
 
   RETVAL_ZVAL(datetime, 0, 0);
 }
 /* }}} */
 
+/* {{{ Cassandra\Date::fromDateTime() */
+PHP_METHOD(Date, fromDateTime)
+{
+  cassandra_date *self;
+  php_date_obj* datetime_obj;
+  timelib_long timestamp;
+  int error;
+
+  if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O",
+                            &datetime_obj,
+                            php_date_get_date_ce()) == FAILURE) {
+    return;
+  }
+
+  if (!datetime_obj->time) {
+    zend_throw_exception_ex(cassandra_invalid_argument_exception_ce, 0 TSRMLS_CC,
+                            "DateTime object has not been correctly initialized");
+    return;
+  }
+  timelib_update_ts(datetime_obj->time, NULL);
+
+  timestamp = timelib_date_to_int(dateobj->time, &error);
```

Comment:
**`timelib_update_ts`** and **`timelib_date_to_int`** is an internal function and is not exported in the PHP dynamic library. Linking the extension statically into the executable is possible; however a DLL is not.  Since these functions are used in both `Time.c` and `Date.c` a utils implementation would be best.